### PR TITLE
fix reference to gateway to ensure deletion of dns and certs

### PR DESCRIPTION
**Pull request by maleck13**
to verify run through the walkthrough and then delete the gateway. 

```
kubectl delete gateway prod-web -n multi-cluster-gateways

```
the dns and tls resources should also be gone

### CODE REVIEW

File: pkg/controllers/gateway/gateway_controller.go [hunk 1]
```diff
@@ -120,35 +120,29 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
+		fmt.Println("upstream post reconcile ", upstreamGateway.UID, upstreamGateway.ResourceVersion)
+		controllerutil.RemoveFinalizer(upstreamGateway, GatewayFinalizer)
 		if err := r.Update(ctx, upstreamGateway); err != nil {
 			return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from gateway : %s", err)
 		}
+		log.Info("gateway being deleted finalizer removed ", "gateway", upstreamGateway.Name, "namespace", upstreamGateway.Namespace)
 		return ctrl.Result{}, nil
 	}
 
+	if !controllerutil.ContainsFinalizer(upstreamGateway, GatewayFinalizer) && !isDeleting(upstreamGateway) {
```
Comment (mikenairn) [inline pkg/controllers/gateway/gateway_controller.go:137]: isDeleting is redundant now, but not an issue either

File: docs/ocm-control-plane-walkthrough.md [hunk 2]
```diff
@@ -223,7 +223,7 @@ spec:
     name: prod-web
     namespace: kuadrant-multi-cluster-gateways
   hostnames:
-  - "sub.replace.this"  
+  - "api.cb.hcpapps.net"
```
Comment (mikenairn) [inline docs/ocm-control-plane-walkthrough.md:226]: revert?

File: pkg/controllers/gateway/gateway_controller.go [hunk 3]
```diff
@@ -207,13 +201,14 @@ func (r *GatewayReconciler) reconcileDownstreamFromUpstreamGateway(ctx context.C
 		downstream.Labels = map[string]string{}
 	}
 	downstream.Labels[ManagedLabel] = "true"
-	accessor := traffic.NewGateway(downstream)
-	if isDeleting(downstream) {
+	accessor := traffic.NewGateway(upstreamGateway)
+	if isDeleting(upstreamGateway) {
```
Comment (mikenairn) [inline pkg/controllers/gateway/gateway_controller.go:205]: How does it get here? Do we not return earlier if this is deleting?

Comment (maleck13) [reply pkg/controllers/gateway/gateway_controller.go:205]: it calls the reconcile in the deleting block

Comment (maleck13) [reply pkg/controllers/gateway/gateway_controller.go:205]: https://github.com/Kuadrant/multicluster-gateway-controller/pull/216/files#diff-6afa7d73841a5c0ebdb78eaf859ce5a400203a037f871440940b5347c3a97376R125

Comment (mikenairn) [reply pkg/controllers/gateway/gateway_controller.go:205]: ignore, i read this as being further down the Reconcile function

**Pull request merged**